staging/src/k8s.io/apiserver/pkg/util/flowcontrol/max_seats.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flowcontrol
+
+import (
+	"sync"
+)
+
+type MaxSeatsTracker interface {
+	// GetMaxSeats returns the maximum seats a request should occupy for a given priority level.
+	GetMaxSeats(priorityLevelName string) uint64
+
+	// SetMaxSeats configures max seats for a priority level.
+	SetMaxSeats(priorityLevelName string, maxSeats uint64)
```

Comment:
ack -- I added a TODO comment under MaxSeatsTracker interface. Fwiw we don't clean up a lot of state in the controller (e.g. plStates) so it might be good to re-consider piority level deletion in the larger context of this controller.

Reply:
Huh?  The controller quite explicitly and carefully considers deletion of priority levels.  Look in the code that digests the config objects.

Reply:
oh my bad -- I was only looking at the delete event handler in the informer. I added delete logic in processOldPLsLocked. Thanks!

Reply:
This looks resolved to me.